architecture/2024-11-authz-feature-spec.md

```diff
@@ -0,0 +1,524 @@
+
+**Operators** – Operators, sometimes referred to as Site Reliability Engineers (SREs) ensure the availability, performance, and security applications running in production. In some cases, the operator persona is combined with the developer and referred to as DevOps. In other cases, the security responsibility is delegated to a dedicated security operator (SecOps).
+
+## Scenarios
+
+### Scenario 1 – Getting started and local development
+
+When a first-time user or a developer installs Radius, the authorization configuration should be wide open to the user. The user should never have to configure anything or be presented with any options. 
+
+#### User story 1 – Running the first application
+
+As a developer, I just installed Radius on my local workstation and I want to run my first application using Radius.
+
+##### Summary
+
+Radius today creates a `default` workspace, with a `default` resource group, and a `default` environment. In order to introduce the resource group concept the  `rad run` and `rad deploy` user experience is changed so that the user is required to specify a resource group. The default environment remains but is placed in a new `env-default` resource group. The full list of changes includes:
+
+* The `scope` property is replaced with `resource-group` in the config.yaml since scope is ambiguous for the user
+* `rad init` is modified:
+  * The default workspace no longer includes a scope property
+  * The out-of-the-box environment is renamed from `default` to the context string from the KUBECONFIG file (the `connection.context` is already set to this value)
+  * The default resource group is renamed `env-default`
+* `rad run` and `rad deploy` error out if a group is not specified in the command line or config.yaml
+* `rad run` and `rad deploy` now create the specified group if it does not exist
+* `rad run` and `rad deploy` respect the role definitions and assignments described in this document
+* The `Creating application...` output from `rad run` is modified to be more precise
+* New `rad workspace update` command
+
+> [!NOTE]
+>
+> The naming convention for resource groups is `app-group-name` for groups container applications and application resources and `env-group-name` for environments and their recipes. "Resource group" is the long name and should be used in the UI and documentation. "Group" is the short name used in CLI commands or other technical contexts.
```

Comment:
For the naming convention, any resource group contain an application needs to also contain the environment name as well. 

If you imagine an app named `shoestore` that's deployed to `prod-east` and `prod-west`, the problem is obvious. These two deployments need to be separate groups, or they collide.

eg:

- `prod-east-shoestore`
- `prod-west-shoestore`